app/models/event.rb

```diff
@@ -150,12 +150,23 @@ def accept_all_application_letters
     end
   end
 
+  # Sets the status_notification_sent flag for all application letters of the given type
+  #
+  # @param status [Type] the desired application status the flag should be set for
+  # @return none
+  def set_status_notification_flag_for_applications_with_status(status)
+    applications = application_letters.select {|application| application.status == status.to_s}
+    applications.each do |application_letter|
```

Comment:
Instead of iterating over the collection I think you could use map! to call a method on every object in a collection. Just a little note. This way it also works and it isn't that bad.